FirebaseInstallations/Source/Library/FIRInstallationsItem.m

```diff
@@ -52,4 +52,30 @@ + (NSString *)identifierWithAppID:(NSString *)appID appName:(NSString *)appName
   return [appID stringByAppendingString:appName];
 }
 
++ (NSString *)generateFID {
+  NSUUID *uuid = [NSUUID UUID];
+  uuid_t uuidBytes;
+  [uuid getUUIDBytes:uuidBytes];
+
+  NSData *uuidData = [NSData dataWithBytes:uuidBytes length:16];
+
+  uint8_t prefix = 0b01110000;
+  NSMutableData *fidData = [NSMutableData dataWithBytes:&prefix length:1];
+
+  [fidData appendData:uuidData];
+  NSString *fidString = [fidData base64EncodedStringWithOptions:0];
+
+  // TODO: Consider implementation which does not modify UUID.
+
+  // A valid FID has exactly 22 base64 characters, which is 132 bits, or 16.5
+  // Our generated ID has 16 bytes UUID + 1 byte prefix which is 23 base64 characters
```

Comment:
16+1=23???

Reply:
That's how base64 encoding works.

Reply:
Updated the explanation a bit. Let me know if it is still not exactly clear.

Reply:
Thanks! That clarifies it.

Reply:
Thanks!